packages/rulesets/src/spectral/functions/tags-not-allowed-in-properties-bag.ts

```diff
@@ -0,0 +1,29 @@
+import _ from "lodash"
+import { deepFindObjectKeyPath, getProperties } from "./utils"
+
+const TAGS = "tags"
+const PROPERTIES = "properties"
+const NestedPROPERTIES = "properties"
+const ERROR_MESSAGE = "Tags should not be specified in the properties bag for proxy resources. Consider using a Tracked resource instead."
+
+export const tagsNotAllowedInPropertiesBag = (definition: any, _opts: any, ctx: any) => {
+  const properties = getProperties(definition)
+  const errors = []
+
+  if ("tags" in properties && !("location" in properties)) {
```

Comment:
>!("location" in properties) [](http://example.com/codeflow?start=30&length=27)

You can either generalize the logic to check for tags in properties for tracked or proxy types

OR

Keep the logic to check only for proxy types and change the name of the rule.

Right now, the rule name is tags are not allowed in properties bag but the logic checks it only for proxy resources.

Reply:
Changed the rule name to reflect only proxy resources.